insights/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,427 @@
+	// +kubebuilder:validation:XValidation:rule=`isURL(self) && url(self).getScheme() == "https"`,message=`URI must be a valid HTTPS URL (e.g., https://example.com)`
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	URI string `json:"uri,omitempty"`
+}
+
+// healthCheck represents an Insights health check attributes.
+type HealthCheck struct {
+	// description is required field that provides basic description of the healtcheck.
+	// It must contain at least 10 characters and may not exceed 2048 characters.
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:MinLength=10
+	// +required
+	Description string `json:"description"`
+	// totalRisk is the required field of the healthcheck.
+	// It is indicator of the total risk posed by the detected issue; combination of impact and likelihood.
+	// Allowed values are Low, Medium, Important and Critical.
+	// The value represents the severity of the issue.
+	// +required
+	TotalRisk TotalRisk `json:"totalRisk"`
+	// advisorURI is required field that provides the URL link to the Insights Advisor.
+	// The link must be a valid HTTPS URL and the maximum length is 2048 characters.
+	// +kubebuilder:validation:XValidation:rule=`isURL(self) && url(self).getScheme() == "https"`,message=`advisorURI must be a valid HTTPS URL (e.g., https://example.com)`
+	// +kubebuilder:validation:MaxLength=2048
+	// +required
+	AdvisorURI string `json:"advisorURI"`
+}
+
+// totalRisk defines the valid totalRisk values.
+// +kubebuilder:validation:Enum=Low;Moderate;Important;Critical
+type TotalRisk string
+
+const (
+	TotalRiskLow       TotalRisk = "Low"
+	TotalRiskModerate  TotalRisk = "Moderate"
+	TotalRiskImportant TotalRisk = "Important"
+	TotalRiskCritical  TotalRisk = "Critical"
+)
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
+	// group is required field that specifies the API Group of the Resource.
+	// Enter empty string for the core group.
+	// This value is empty or it should follow the DNS1123 subdomain format.
+	// It must be at most 253 characters in length, and must consist only of lower case alphanumeric characters, '-' and '.', and must start and end with an alphanumeric character.
+	// Example: "", "apps", "build.openshift.io", etc.
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || !format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +kubebuilder:validation:MaxLength:=253
+	// +required
+	Group string `json:"group"`
+	// resource is required field of the type that is being referenced.
+	// It is normally the plural form of the resource kind in lowercase.
+	// It must be at most 63 characters in length, and must must consist of only lowercase alphanumeric characters and hyphens
```

Comment:
And must start with an alphabetic character and end with an alphanumeric character